### PR TITLE
Define `translated_name` in `OrderLineDiscount` for promotions

### PR DESCRIPTION
**Pull request by IKarbowiak**
Add a promotion translated name to `OrderLineDiscount`

Part of https://github.com/saleor/saleor/issues/13316

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migrations are either absent or optimized for zero downtime
* [ ] The changes are covered by test cases

### CODE REVIEW

**Review by SzymJ — APPROVED** [review]

File: saleor/discount/tests/test_discounts.py [hunk 1]
```diff
@@ -557,3 +560,109 @@ def test_fetch_catalogue_info_for_sale_has_one_element_sets(sale):
...
+    name = get_discount_name(rule, promotion)
+
+    # then
+    assert name == ""
+
+
+def test_get_discount_translated_name_only_rule_translation(rule_info):
+    # given
+    rule_info_data = rule_info._asdict()
+    rule_info_data["promotion_translation"] = None
```
Comment (aniav) [inline saleor/discount/tests/test_discounts.py:623]: Hmm, shouldn't
```
rule_info.promotion_translation = None
```

do the same thing? I might not understand how `promotion_translation` value is filled and trying to grasp that.

**Review by aniav — APPROVED** [review]
Two small comments, but nothing blocking from merging 🙌

File: saleor/discount/tests/test_discounts.py (hunk 1 above)
Comment (IKarbowiak) [reply saleor/discount/tests/test_discounts.py:623]: `rule_info` is `NamedTuple` so it's immutable and I cannot do it in simpler way :(

File: saleor/graphql/checkout/tests/mutations/test_checkout_complete.py [hunk 2]
```diff
@@ -1261,6 +1269,9 @@ def test_checkout_with_voucher_complete_product_on_promotion(
         pk=checkout.pk
     ).exists(), "Checkout should have been deleted"
 
+    with pytest.raises(checkout_line_discount._meta.model.DoesNotExist):
+        checkout_line_discount.refresh_from_db()
+
```
[the remark this answers is not in the document]
Comment (IKarbowiak) [reply saleor/graphql/checkout/tests/mutations/test_checkout_complete.py:1274]: After checkout completing, the checkout related objects are removed, and that's why I'm checking if this object does not exists anymore. Instead `OrderLineDiscount` object is created

**Pull request merged**